openmc/mgxs/mgxs.py

```diff
@@ -1335,18 +1457,44 @@ def print_xs(self, subdomains='all', nuclides='all', xs_type='macro'):
                 string += '{0: <16}\n'.format(xs_header)
                 template = '{0: <12}Group {1} [{2: <10} - {3: <10}eV]:\t'
 
-                # Loop over energy groups ranges
-                for group in range(1, self.num_groups+1):
-                    bounds = self.energy_groups.get_group_bounds(group)
-                    string += template.format('', group, bounds[0], bounds[1])
-                    average = self.get_xs([group], [subdomain], [nuclide],
-                                          xs_type=xs_type, value='mean')
-                    rel_err = self.get_xs([group], [subdomain], [nuclide],
-                                          xs_type=xs_type, value='rel_err')
-                    average = average.flatten()[0]
-                    rel_err = rel_err.flatten()[0] * 100.
-                    string += '{:.2e} +/- {:1.2e}%'.format(average, rel_err)
-                    string += '\n'
+                average_xs = self.get_xs(nuclide=[nuclide],
+                                         subdomain=[subdomain],
+                                         xs_type=xs_type, value='mean')
+                rel_err_xs = self.get_xs(nuclide=[nuclide],
+                                         subdomain=[subdomain],
+                                         xs_type=xs_type, value='rel_err')
+                rel_err_xs = rel_err_xs * 100.
+
+                if self.num_polar or self.num_azimuthal:
```

Comment:
On a separate note, this PR is a case in point as to why OO programming with as little redundant code as possible is so valuable. It's not your fault, but we have a LOT of redundant code between "mgxs.py" and "mdgxs.py" which makes extensions like yours more challenging that it would otherwise be with if the two modules were less redundant to begin with.